public/app/percona/pmm-dump/PMMDump.tsx

```diff
@@ -0,0 +1,328 @@
+import { PMMDumpService } from './PMMDump.service';
+import { getStyles } from './PmmDump.styles';
+import { SendToSupportModal } from './SendToSupportModal';
+import { PmmDumpLogsModal } from './components/PmmDumpLogsModal/PmmDumpLogsModal';
+export const NEW_BACKUP_URL = '/pmm-dump/new';
+
+const pageNav: NavModelItem = {
+  icon: 'brain',
+  id: 'user-new',
+  text: 'PMM Dump',
+  subTitle:
+    'Simplify troubleshooting and accelerate issue resolution by securely sharing relevant data, ensuring a smoother support experience.',
+};
+
+export const PMMDump = () => {
+  const styles = useStyles2(getStyles);
+  const dispatch = useAppDispatch();
+  const { dumps } = useSelector(getDumps);
+  const [triggerTimeout] = useRecurringCall();
+  const [selected, setSelectedRows] = useState<Array<Row<PMMDumpServices>>>([]);
+  const [selectedDumpIds, setSelectedDumpIds] = useState<string[]>([]);
+  const [selectedDump, setSelectedDump] = useState<PMMDumpServices | null>(null);
+  const [isSendToSupportModalOpened, setIsSendToSupportModalOpened] = useState(false);
+  const [logsModalVisible, setLogsModalVisible] = useState(false);
+
+  const loadData = useCallback(async () => {
+    try {
+      await dispatch(fetchPmmDumpAction());
+    } catch (e) {
+      if (isApiCancelError(e)) {
+        return;
+      }
+      logger.error(e);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getLogs = useCallback(
+    async (startingChunk: number, offset: number, token?: CancelToken) => {
+      return PMMDumpService.getLogs(selectedDump?.dump_id || '', startingChunk, offset, token);
+    },
+    [selectedDump]
+  );
+
+  useEffect(() => {
+    loadData().then(() => triggerTimeout(loadData, DATA_INTERVAL));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadData]);
+
+  const closeEditModal = (saved = false) => {
+    setIsSendToSupportModalOpened(false);
+    setSelectedDump(null);
+  };
+
+  const getActions = useCallback(
+    (row: Row<PMMDumpServices>): Action[] => [
+      {
+        content: (
+          <HorizontalGroup spacing="sm">
+            <Icon name="trash-alt" />
+            <span className={styles.actionItemTxtSpan}>{Messages.services.actions.delete}</span>
+          </HorizontalGroup>
+        ),
+        action: () => {
+          onDelete(row.original);
+        },
+      },
+      {
+        content: (
+          <HorizontalGroup spacing="sm">
+            <Icon name="arrow-right" />
+            <span className={styles.actionItemTxtSpan}>{Messages.services.actions.sendToSupport}</span>
+          </HorizontalGroup>
+        ),
+        action: () => {
+          setSelectedDumpIds([row.original.dump_id]);
+          setIsSendToSupportModalOpened(true);
+        },
+      },
+      {
+        content: (
+          <HorizontalGroup spacing="sm">
+            <Icon name="eye" />
+            <span className={styles.actionItemTxtSpan} onClick={() => onLogClick(row.original)}>
```

Comment:
Any reason to have the log open action here and not in `action`?

Reply:
moved